ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -75,6 +75,53 @@ const handleTieredSearchLocalFinish = function(ids) {
   this.startSearch({ results, status })
 }
 
+const reducer = (state = [{}], action) => {
+  switch (action.type) {
+    case 'CLEAR_PAGES':
+      return [{}]
+    case 'NEXT_PAGE':
+      return state.concat({})
+    case 'PREVIOUS_PAGE':
+      return state.slice(0, -1)
+    case 'UPDATE_RESULTS':
+      const srcs = action.payload.results
+        .map(({ src }) => src)
+        .reduce((counts, src) => {
+          if (counts[src] === undefined) {
+            counts[src] = 0
+          }
+          counts[src] += 1
+          return counts
+        }, {})
+
+      return state.slice(0, -1).concat(srcs)
+    default:
+      return state
+  }
+}
+
+const currentIndexForSource = state => {
+  return state.reduce((counts, page) => {
+    return Object.keys(page).reduce((counts, src) => {
+      const resultCount = page[src]
+
+      if (counts[src] === undefined) {
+        counts[src] = 1
```

Comment:
counts[src] = 0 ? could you use a named function here to make things more readable?

Reply:
We are starting the count at 1 if the src is found.